docs/DeveloperGuide.md

```diff
@@ -499,23 +508,42 @@ todo
 
 { end of `implementation#exporting_data` written by: ______________ }
 
-{ start of `implementation#data_encryption` written by: ______________ }
+{ start of `implementation#data_encryption` written by: Marcus Tan Wei }
 
-### 4.10. [Proposed] Data encryption
+### 4.10 [Proposed] Data encryption
+Data encryption can be implemented in the future versions. This is to further protect the students' particulars. 
 
-todo
+### 4.10.1 Solution 1
+Encrypt and store data locally.
+
+* Pros: 
+   * Easy to implement. One example is through the usage of Java Cryptography API.
+
+* Cons: 
+   * Performance reduction may occur since every call to save to storage requires encrypting a lot of data.
+   * Security issues may still arise if we store the encryption key in the same machine.
 
-{ end of `implementation#dat_encryption` written by: ______________ }
+### 4.10.2 Solution 2
+Store data outside the user’s machine and issue them access token.
+
+* Pros: 
+   * Data will not be only bounded to one machine, hence, can be restored if the user forgets their credentials.
```

Comment:
Perhaps "Data will be bounded outside of one machine/bounded to other machines or servers" may flow better?
Again, should it be either "user" and "his/her" or "users" and "their"?

Reply:
Thank you for the suggestion!